functionsandLoops/funcandloops_ex.py

```diff
@@ -1,8 +1,79 @@
+# ex1 - the function prints all the numbers from 1 to n in seperate lines and print the function's return value
 def func(n: int):
     for i in range(n):
         print (i)
     return "finished"
 
+# ex2 - function which prints 3 different sequences
+def fun(n: int):
+    # first required sequence
+    for i in range(n+1):
+        # converting the first element into '1' 
+        if i == 0:
+            print("First sequence:\n", 1, end = " ")
+        # the other elements are the even ones 
+        if i%2 == 0 and i != 0:
+            print(i, end = " ")
+    print ("\n", end = " ")
+    # second required sequence 
+    for i in range(n+1):
+        # converting the first element into '1' 
+        if i == 0:
+            print("Second sequence:\n", 1, end = " ")
+        # the other elements are the powerset of 2
+        elif 2**i <= (n+1):
+            print(2**i, end = " ")
```

Comment:
You did good in creating an a(i) which represents the series (in this case 2**i).
But your if works EXTRA time. try to make your function STOP working after reaching 2**i > n.
in the current state. the function will work n if there is no need to keep calculating.